src/Mod/PartDesign/Gui/ReferenceSelection.cpp

```diff
@@ -86,13 +92,19 @@ bool ReferenceSelection::allow(App::Document* pDoc, App::DocumentObject* pObj, c
             return false;
     }
 #endif
-    // Handle selection of geometry elements
-    if (Base::Tools::isNullOrEmpty(sSubName))
+    if (!sSubName || sSubName[0] == '\0'){
+        if (pObj->isDerivedFrom<Sketcher::SketchObject>()) {

```

Comment:
You can loosen up the direct dependency by calling the string version instead:
```suggestion
        if (pObj->isDerivedFrom("Sketcher::SketchObject")) {
```

Reply:
@ziad-ashraf7 this should be incorporated into your code. It will remove the direct Sketcher dependency. Then you can strip out all of those new includes and CMake changes.